File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prefix = []
        suffix = []
        product = 0

        for i in range(len(nums)):
            if i == 0:
                product = 1
            else:
                product*= nums[i-1]

            prefix.append(product)

        for i in range(len(nums) -1, -1, -1):
            if i == len(nums) -1:
                product = 1

            else:
                product *= nums[i+1]

            suffix.insert(0, product)

        results = []

        for i in range(len(nums)):
            result = prefix[i] * suffix[i]
            results.append(result)

        return results                   

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        if(nums.size() == 0){
            return 0;
        }

        if(nums.size() == 1){
            return (nums[0] == k);
        }

        int result = 0;
        int r = 0;
        int current_sum = 0;

        for(int i = 0; i<nums.size(); ++i){

            if(i == nums.size() -1){
                if(nums[i] == k){
                    result++;
                }
                break;
            }

            if(nums[i] == k){
                result++;
            }

            current_sum = nums[i];
            r = i+1;

            while(r<nums.size()){
                current_sum += nums[r];
                if(current_sum == k){
                    result++;
                }
                r++;
            }

        }

        return result;
    }
};